src/inmanta/data/__init__.py

```diff
@@ -3322,7 +3322,7 @@ async def get_compile_details(cls, environment: uuid.UUID, id: uuid.UUID) -> Opt
                     {cls.table_name()} comp
                     INNER JOIN compiledetails cd ON cd.substitute_compile_id = comp.id
                     LEFT JOIN public.report rep on comp.id = rep.compile
-        ) SELECT * FROM compiledetails;
+        ) SELECT * FROM compiledetails ORDER BY report_started ASC;
```

Comment:
Did you do run an `EXPLAIN ANALYZE` on this query? This sort operation might have a rather big performance impact.

Reply:
After your question I did :)
The query only deals with one specific compile (or a few, if there were substitutions), so it has a quite limited scope. The sorting doesn't seem to affect the performance much:
Without sorting: https://explain.dalibo.com/plan/no6#plan
With sorting: https://explain.dalibo.com/plan/poA

Reply:
Indeed. The query plan indicates that the cost of the sorting is high. But in practice the number of records returns by the query should be very small.

Reply:
Does the performance improve if we add an index on `Report(compile, report_started ASC)`?

Reply:
It does not, and I think it's because we are sorting the records of the CTE. 
(For building the CTE, the `report_compile_index` is used).
On the other hand, the number of records we're dealing with here should be small indeed.